src/main/java/com/gojek/beast/models/Record.java

```diff
@@ -15,10 +15,14 @@
     private Map<String, Object> columns;
 
     public String getId() {
-        return String.format("%s_%d_%d", offsetInfo.getTopic(), offsetInfo.getPartition(), offsetInfo.getOffset());
+        return String.format("%s_%d_%d", offsetInfo.getTopic(), getPartition(), offsetInfo.getOffset());
+    }
+
+    public Integer getPartition() {
+        return offsetInfo.getPartition();
     }
 
     public long getSize() {
-        return columns.toString().getBytes(StandardCharsets.UTF_8).length;
+        return columns == null ? 0 : columns.toString().getBytes(StandardCharsets.UTF_8).length;
```

Comment:
@mauliksoneji Also, I wanted to confirm if this could create a problem anywhere? I think `columns` attribute is usually never empty except for the tests right?

Reply:
yes you are correct